### ISSUE TITLE
Update the Breadcrumbs component in the VAOS application

### ISSUE THREAD

**Issue by simiadebowale** [department-of-veterans-affairs/va.gov-team#40139]
Platform Design System team built a va-breadcrumbs component. We should replace the existing Breadcrumbs component with the updated va-breadcrumbs component. There is a migration script for the component so it can be easily migrated from the React to the web component version.

[Breadcrumbs Design component](https://design.va.gov/components/breadcrumbs)
[Breadcrumbs storybook](https://design.va.gov/storybook/?path=/docs/components-va-breadcrumbs--default)

---
## Known Differences

Tasks:
- Try out replacing the Breadcrumbs component with va-breadcrumbs component and document any differences that appear

## Acceptance Criteria
- [x] Replace Breadcrumbs components with va-breadcrumbs web components

---

**Comment by ldelacosta** [comment]
@ryanshaw - In the mobile view, there is a spacing issue when the user starts to go to the scheduling workflow. Here are the [current design specs](https://www.figma.com/file/jkja09Pi1pENB0nDZ7usFm/Spacing-issue) with notes from Peter. If this is a separate issue, please let me know.

**Comment by humancompanion-usds** [comment]
@ryanshaw, @ldelacosta, or @outerpress -  I don't have access to the Figma design specs. Can you tell us if this problem is the same as this [issue](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/469)? If so, we have a PR that will be merged this sprint, though I guess that has come a little too late. Following that thread, was their a support request to the Design System Team on this issue? Thanks.

**Comment by ryanshaw** [comment]
@humancompanion-usds @ldelacosta @outerpress I looked into this and the two issues are not related but we are experiencing the wrapping issue currently. The comments on this issue were related to indentation that was caused by moving to the web component version of the breadcrumbs. That has been temporarily reverted.

![image](https://user-images.githubusercontent.com/587812/175666464-b097f7d7-9b30-4f06-a21f-37f91da10eea.png)

**Comment by outerpress** [comment]
I see three issues in the VAOS breadcrumb:

1. Too much space above and below breadcrumb (compared to VA.gov - this is what [Leah mentioned earlier](https://github.com/department-of-veterans-affairs/va.gov-team/issues/40139#issuecomment-1146213937))
2. Text breaks away from the "‹" when breadcrumb content reaches a certain length - this is what [#469](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/469) covers.
3. Breadcrumb left margin doesn't line up with the left margin on the rest of the page, which I think is what Ryan called out above.

Staging isn't letting me login to confirm the current reality, after reverting changes. I'll double-check later.

@humancompanion-usds you should have access to the [Figma file](https://www.figma.com/file/jkja09Pi1pENB0nDZ7usFm/Spacing-issue?node-id=2%3A4) now (not really a design spec, just an illustration of problem 1).

**Comment by ryanshaw** [comment]
@k80bowman I'm currently working on this and couldn't previously find an app that was using it, so I'll take a look at that. Thanks for that info.

**Comment by k80bowman** [comment]
@ryanshaw I just searched for `va-breadcrumbs` in the vets-website applications directory. There were several examples that might help you out.

**Comment by ryanshaw** [comment]
It actually looks like that app is using the [older React component](https://github.com/department-of-veterans-affairs/component-library/tree/master/packages/react-components/src/components/Breadcrumbs): 
`import Breadcrumbs from '@department-of-veterans-affairs/component-library/Breadcrumbs';`

We are attempting to migrate to the [new web component](https://design.va.gov/components/breadcrumbs) and import like this: 
`import { VaBreadcrumbs } from '@department-of-veterans-affairs/component-library/dist/react-bindings';`

I did a little digging and there are two ways to use the new web components, one way is the way we are doing it now which adds React bindings so the component can accept more types of properties.

> If the Design System web components will be used in a React application, you are ready to use them unless:
>
>- You must pass in a function, object or array to a web component’s properties
>- You must use custom events

With our current implementation we don't need to do that so I changed the way we are using it and it looks like the spacing changes:

Before:
![image](https://user-images.githubusercontent.com/587812/176273054-ae2ebca2-6a4d-49dc-b43f-fb36c6f3ffc0.png)

After: 
![image](https://user-images.githubusercontent.com/587812/176273198-c4c3a4c4-2511-4f3a-ba8c-4123290ea8f6.png)

**Comment by k80bowman** [comment]
Oh interesting. It would really be helpful to have a link where we can inspect your code in the browser so we can see what's going on.

And that app is using our new web component, if you inspect their breadcrumbs you'll see this:
![image](https://user-images.githubusercontent.com/12739849/176277336-b20269be-2a5c-4f13-8b7b-cfc587be41ad.png)

And you can see the [code here](https://github.com/department-of-veterans-affairs/vets-website/blob/55a244c11ce91f8d9d18d13fcc875e935d07d2a3/src/applications/disability-benefits/all-claims/utils/index.jsx#L706-L714).

**Comment by ryanshaw** [comment]
Ah interesting. I had just done a really quick search and was looking [here](https://github.com/department-of-veterans-affairs/vets-website/blob/55a244c11ce91f8d9d18d13fcc875e935d07d2a3/src/applications/disability-benefits/2346/containers/App.jsx#L2).

Can you view review instances? I'll need to revert a commit but can get that up pretty quickly.

**Comment by k80bowman** [comment]
I can view review instances, I'll just need any details on signing in. You can send them in Slack if you don't want to post them here.

**Comment by ryanshaw** [comment]
@k80bowman I just pushed that change but it will take a few minutes to build. I will update once the build is finished.

**Comment by ryanshaw** [comment]
Without the `className` prop: 

![image](https://user-images.githubusercontent.com/587812/176543395-3fb5a3aa-b764-4d5a-b6ad-fe926cbe7b70.png)

With the `className` prop:

![image](https://user-images.githubusercontent.com/587812/176543594-62443ebc-6367-4467-9779-86f1b1970ecd.png)

_(The review instance is returning a bad gateway at the moment, so this is just locally...)_

**Comment by k80bowman** [comment]
Thank you so much for troubleshooting this with me! I think I’ve figured out the problem, but I’m not sure why it’s a problem. The extra spacing is coming from a `vaos-appts__breadcrumbs` class that is being added via the `className` prop. It's adding 2em padding to the top and bottom of the component. The web component version of the Breadcrumbs component does not have a className prop (the older React version does), so when using the regular web component version I’m assuming, since you’re not seeing the problem, it just ignores it. For some reason, though, when using the React bindings version, it isn't ignoring that prop. I’ll ask my team and see if this is something they’ve seen before, but I certainly haven’t.

![image](https://user-images.githubusercontent.com/12739849/176546999-a605a6f5-4597-4875-a723-824180316226.png)
![image](https://user-images.githubusercontent.com/12739849/176547026-700c5456-32f4-4d6f-8946-6603cfaa1795.png)

**Comment by ryanshaw** [comment]
For reference, here's the spacing without the class (and using the React version of the component). I believe the class was from when the breadcrumbs were created specifically for and within the VAOS app -- does not look like it is needed any longer. 

![image](https://user-images.githubusercontent.com/587812/176554875-e01de79e-eb64-4cfb-b1a9-1a45824a076f.png)

**Comment by ldelacosta** [comment]
@simiadebowale - Can we check with the Shared Support team to see if they can help validate this ticket?

**Comment by tlei123** [comment]
From Leah via Slack, test-user for validation in Staging is **patient997@id.me** [test-user passwords are [here](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/Administrative/vagov-users/staging-test-accounts-vaos.md)].

**Comment by tlei123** [comment]
@ryanshaw , the **AC on _this_ ticket** [Replace Breadcrumbs components with va-breadcrumbs] has been validated satisfied on Staging, incl. all the scheduling form-pages, so I've checked that off above in the main description. 

The va-breadcrumbs component _padding_ looks good on Mac Chrome, but I'm including my window-resize video below to show the _line-spacing_ once wrapping starts, just in case you think that's an issue with the va-breadcrumbs component:

https://user-images.githubusercontent.com/587583/181399583-f5c8a296-0564-4b04-9132-6b0b71642360.mp4

Re. the **[Design-System issue](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/469)** mentioned in the conversation above:
- **320px** viewport-width: 
  - Mac Chrome mobile-emulation &mdash; The new va-breadcrumb component's padding looks good to me: 
        ![image](https://user-images.githubusercontent.com/587583/181400855-da08e426-901d-4a70-858e-0e50fd5bd9a1.png)
  - BrowserStack.com &mdash; No 320px-viewport-width device could reach Staging _authenticated_ homepage &mdash; either the unauth'd homepage wouldn't finish loading and errored out, or the device itself wouldn't load at all on BrowserStack.  That's probably because all 320px-viewport-width devices are 4+ years old &mdash I'm guessing the issue-opener discovered this on desktop using mobile _emulation_, and am satisfied that this use-case is sufficiently edge.
- 360px viewport-width &mdash; The new va-breadcrumb component's padding looks good to me, both on Samsung S21 Chrome & Apple iPhone 12 Safari: 
    ![image](https://user-images.githubusercontent.com/587583/181402483-1dd16abd-0ccb-4e72-a7c1-47fb2868d1a4.png) 
    ![image](https://user-images.githubusercontent.com/587583/181403205-02e54eaf-641c-4ca6-934b-46c1f3efc40e.png)

Let me know if you need further validation, but I think for CSS the above is sufficient.

cc @ldelacosta

**Comment by simiadebowale** [comment]
@tlei123 I also wanted to confirm that navigation through the breadcrumb component is working as expected with no console errors. I.e clicking on 'Schedule and manage health appointments'  - should take you back to this url: https://staging.va.gov/health-care/schedule-view-va-appointments/appointments/ etc
thank you.

**Comment by simiadebowale** [comment]
@tlei123 I also wanted to confirm that navigation through the breadcrumb component is working as expected with no console errors. I.e clicking on 'Schedule and manage health appointments'  - should take you back to this url: https://staging.va.gov/health-care/schedule-view-va-appointments/appointments/ etc
thank you.

**Comment by tlei123** [comment]
Thx @simiadebowale !  If you, @ldelacosta , or @ryanshaw want(s) to have Design Systems look into that big line-spacing (when the component wraps), feel free to link my comment above into their [existing ticket](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/469) [or a create a new tic over there].  I'm satisfied the wrapping use-case is edgy enough not to open a bug.

**Comment by simiadebowale** [comment]
LGTM, Thank you! @tlei123
